src/client/Microsoft.Identity.Client/ApiConfig/DownStreamRestApi/PublicClientDownstreamApi.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.Logger;
+using Microsoft.IdentityModel.Abstractions;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Down stream api for public clients
+    /// </summary>
+    public class PublicClientDownstreamApi
+    {
+        private readonly int _maximumAuthenticationAttempts = 3;
+        string _resourceName;
+        DownstreamRestApiOptions _options;
+        IPublicClientApplication _pca;
+        private string _currentNonce = null;
+        //TODO, grab identity logger from application.
+        IIdentityLogger _logger = NullIdentityModelLogger.Instance;
+        private int _currentAuthetnicatinonAttempts = 0;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="resourceName"></param>
+        /// <param name="options"></param>
+        public PublicClientDownstreamApi(string resourceName, DownstreamRestApiOptions options)
+        {
+            _resourceName = resourceName;
+            _options = options;
+            _pca = PublicClientApplicationBuilder.CreateWithApplicationOptions(_options).Build();
+            //_logger = options.logger.
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="targetUrl"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<HttpResponseMessage> CallGetApiAsync(Uri targetUrl, CancellationToken cancellationToken = default)
+        {
+            bool isPopSupportedByResource = false;
+
+            if (_pca.IsProofOfPossessionSupportedByClient())
+            {
+                AuthenticationHeaderParser parsedHeaders = null;
+                try
+                {
+                    parsedHeaders = await AuthenticationHeaderParser.ParseAuthenticationHeadersAsync(targetUrl.AbsoluteUri).ConfigureAwait(false);
```

Comment:
You don't even try to get a token and send it to the web API? (in case there is no server nonce)?